src/main/java/org/opentripplanner/routing/algorithm/strategies/TransitOnlyEuclideanRWH.java

```diff
@@ -0,0 +1,128 @@
+package org.opentripplanner.routing.algorithm.strategies;
+
+import org.opentripplanner.common.geometry.SphericalDistanceLibrary;
+import org.opentripplanner.routing.algorithm.GenericDijkstra;
+import org.opentripplanner.routing.algorithm.TraverseVisitor;
+import org.opentripplanner.routing.core.RoutingRequest;
+import org.opentripplanner.routing.core.State;
+import org.opentripplanner.routing.edgetype.FreeEdge;
+import org.opentripplanner.routing.edgetype.StreetTransitLink;
+import org.opentripplanner.routing.graph.Edge;
+import org.opentripplanner.routing.graph.Vertex;
+import org.opentripplanner.routing.vertextype.TransitStationStop;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * A Euclidean remaining weight strategy that takes into account transit boarding costs where applicable.
+ */
+public class TransitOnlyEuclideanRWH implements RemainingWeightHeuristic {
+
+    private static final Logger LOG = LoggerFactory.getLogger(TransitOnlyEuclideanRWH.class);
+
+    private double lat;
+    private double lon;
+    private double maxTransitSpeed;
+    private double requiredWalkDistance;
+    private double walkMultiplier;
+    private double streetWeight;
+
+    @Override
+    public void initialize(RoutingRequest options, long abortTime) {
+        Vertex destination = getDestination(options.rctx.target);
+        lat = destination.getLat();
+        lon = destination.getLon();
+
+        requiredWalkDistance = determineRequiredWalkDistance(options);
+        maxTransitSpeed = options.getTransitSpeedUpperBound();
+        walkMultiplier = options.routingReluctances.getWalkReluctance() / options.walkSpeed;
+        streetWeight = walkMultiplier * requiredWalkDistance;
+    }
+
+    private Vertex getDestination(Vertex target) {
+        if (target.getDegreeIn() == 1) {
+            Edge edge = target.getOutgoing().stream().findFirst().get();
+            if (edge instanceof FreeEdge) {
+                return edge.getFromVertex();
+            }
+        }
+        return target;
+    }
+
+    /**
+     * On a transit trip, there are two cases:
+     * (1) we're not on a transit vehicle. In this case, there are two possible ways to compute
+     * the remaining distance, and we take whichever is smaller:
+     * (a) walking distance / walking speed
+     * (b) boarding cost + transit distance / transit speed (this is complicated a bit when
+     * we know that there is some walking portion of the trip).
```

Comment:
z tego co widze to nie bierze pod uwagi dystansu od obecnego miejsca do przystanku. Tylko od końcowego przystanku do celu?

Reply:
Generally this class finds actual distance from closest transit stop to destination. I didn't edit this code at all, because I am not sure if it even works properly right now, as we are using much faster transit heuristic.